halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/KubernetesDistributedService.java

```diff
@@ -426,6 +405,36 @@ default KubernetesContainerDescription buildContainer(String name, ServiceSettin
     return container;
   }
 
+  default void applyCustomSize(KubernetesContainerDescription container, DeploymentEnvironment deploymentEnvironment, String componentName) {
+    Map<String, Map> componentSizing = deploymentEnvironment.getCustomSizing().get(componentName);
+
+    if (componentSizing != null) {
+
+      if (componentSizing.get("requests") != null) {
+        container.setRequests(retrieveKubernetesResourceDescription(componentSizing, "requests"));
+      }
+
+      if (componentSizing.get("limits") != null) {
+        container.setLimits(retrieveKubernetesResourceDescription(componentSizing, "limits"));
+      }
+    }
+
+    /* TODO(lwander) this needs work
+      SizingTranslation.ServiceSize serviceSize = sizingTranslation.getServiceSize(deploymentEnvironment.getSize(), service);
+    */
+  }
+
+  default KubernetesResourceDescription retrieveKubernetesResourceDescription(Map<String, Map> componentSizing, String resourceType) {
+    KubernetesResourceDescription requests = new KubernetesResourceDescription();
+    requests.setCpu(stringOrNull(componentSizing.get(resourceType).get("cpu")));
+    requests.setMemory(stringOrNull(componentSizing.get(resourceType).get("memory")));
+    return requests;
+  }
+
+  // This is super-goofy. What can we do differently for this? Anything in the config parsing logic?
+  default String stringOrNull(Object value) {
+    return value != null ? String.valueOf(value) : null;
+  }
```

Comment:
@lwander We'd be interested in your thoughts on this little nugget. The issue arises when the size specification is in form with only numeric characters: ```"1"``` as opposed to ```"1000m"``` or ```"1Gi"```.

We used ```String.valueOf``` to ensure that those cases were parsed out of the config as strings. That, in turn, caused problems when a value was unset (null), because ```String.valueOf(null)``` evaluates to ```"null"``` as a string, which is _not_ what we want to be in place when something isn't set.

Our tactical solution was this ```stringOrNull``` method, which just feels weird. Is there something more framework-y in the wider halyard code base that we might leverage for this? Or a natural util-ish place we could stick this little method instead of implementing it twice in the deployment code? We also considered putting it in the CustomSizing class and calling that from both places where it's needed.

Reply:
Yeah agreed this is a little funky - but this is also the first time where the type isn't explicit in the config. Would it be possible to make it clear that component sizing is mapping `string -> string -> string`? Or is the fear that this is too restrictive?

Reply:
We were concerned about being too restrictive, but at this point it probably makes sense to explicitly make it all maps with strings.